python_modules/dagster/dagster/_core/definitions/remote_asset_graph.py

```diff
@@ -231,19 +231,22 @@ def from_repository_handles_and_external_asset_nodes(
         repo_node_pairs_by_key: Dict[
             AssetKey, List[Tuple[RepositoryHandle, "ExternalAssetNode"]]
         ] = defaultdict(list)
+        # Build the dependency graph of asset keys.
+        upstream: Dict[AssetKey, Set[AssetKey]] = defaultdict(set)
+        downstream: Dict[AssetKey, Set[AssetKey]] = defaultdict(set)
+
         for repo_handle, node in repo_handle_external_asset_nodes:
             repo_node_pairs_by_key[node.asset_key].append((repo_handle, node))
-
-        # Build the dependency graph of asset keys.
-        all_keys = {node.asset_key for _, node in repo_handle_external_asset_nodes}
-        upstream: Dict[AssetKey, Set[AssetKey]] = {key: set() for key in all_keys}
-        downstream: Dict[AssetKey, Set[AssetKey]] = {key: set() for key in all_keys}
-        for _, node in repo_handle_external_asset_nodes:
             for dep in node.dependencies:
                 upstream[node.asset_key].add(dep.upstream_asset_key)
                 downstream[dep.upstream_asset_key].add(node.asset_key)
+
```

Comment:
collapsed some other iteration while i was here - don't  think this is as critical

Reply:
not sure about ^^, this will produce an empty set later for access to a missing key which we don't want